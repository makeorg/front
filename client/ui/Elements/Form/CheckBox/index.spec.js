import React from 'react';
import snapshotDiff from 'snapshot-diff';
import renderer from 'react-test-renderer';
import { CheckBox } from './index';

jest.mock('Client/ui/Elements/Form/Styled/CheckBox', () => ({
  FakeCheckboxInputStyle: 'FakeCheckboxInputStyle',
  CheckboxLabelStyle: 'CheckboxLabelStyle',
  CheckboxWrapper: 'CheckboxWrapper',
}));

jest.mock('Client/ui/Elements/Form/Styled/Icons', () => ({
  CheckboxIconStyle: 'CheckboxIconStyle',
}));

describe('CheckBox', () => {
  const id = 'foo';
  const name = 'bar';
  const label = 'baz';

  it('must return the diff between snapshot when checkbox is required or optionnal', () => {
    const RequiredCheckbox = renderer
      .create(<CheckBox id={id} name={name} label={label} required />)
      .toJSON();
    const OptionnalCheckbox = renderer
      .create(<CheckBox id={id} name={name} label={label} />)
      .toJSON();
    expect(snapshotDiff(RequiredCheckbox, OptionnalCheckbox)).toMatchSnapshot();
  });

  it('must return the diff between snapshot when checkbox has a defined value or not', () => {
    const CheckboxWithValue = renderer
      .create(<CheckBox id={id} name={name} label={label} value="qux" />)
      .toJSON();
    const DefaultCheckbox = renderer
      .create(<CheckBox id={id} name={name} label={label} />)
      .toJSON();
    expect(snapshotDiff(CheckboxWithValue, DefaultCheckbox)).toMatchSnapshot();
  });
});
