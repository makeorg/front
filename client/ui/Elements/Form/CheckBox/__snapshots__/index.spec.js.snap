// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox must return the diff between snapshot when checkbox has a defined value or not 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -6,11 +6,10 @@
      id="bar"
      name="bar"
      onChange={[Function anonymous]}
      required={false}
      type="checkbox"
-     value="qux"
    />
    <CheckboxLabelStyle
      htmlFor="bar"
      isBlack={false}
      noFontSizeChange={false}
`;

exports[`CheckBox must return the diff between snapshot when checkbox is required or optionnal 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -4,11 +4,11 @@
      checked={false}
      hidden={true}
      id="bar"
      name="bar"
      onChange={[Function anonymous]}
-     required={true}
+     required={false}
      type="checkbox"
    />
    <CheckboxLabelStyle
      htmlFor="bar"
      isBlack={false}
`;
